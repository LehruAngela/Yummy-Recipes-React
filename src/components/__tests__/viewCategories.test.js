import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ViewCategories from '../viewCategories';

describe('ViewCategories', () => {
    const wrapper = shallow(<ViewCategories />);
    const preventDefault = jest.fn();

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
      }); 

    it('handles ViewCategories', () => {
        expect(wrapper.instance().handleViewCategory({ preventDefault }));
    });
    
    it('handles ViewCategories', () => {
        expect(wrapper.instance().handleDeleteCategory({ preventDefault }));
    });

    it('handles ViewCategories', () => {
        expect(wrapper.instance().handlePageChange({ preventDefault }));
    });
 
});