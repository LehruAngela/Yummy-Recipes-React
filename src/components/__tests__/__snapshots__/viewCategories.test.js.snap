// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewCategories renders properly 1`] = `
<div>
  <Component />
  <div>
    <h2
      className="heading"
    >
      Categories
    </h2>
    <CreateCategory
      handleViewCategory={[Function]}
    />
  </div>
  <div>
    <form
      class="form-inline my-2 my-lg-0"
      onSubmit={[Function]}
    >
      <input
        aria-label="Search"
        class="form-control mr-sm-2 search"
        name="q"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
      <button
        className="btn my-2 my-sm-0"
        type="submit"
      >
        <i
          class="fas fa-search"
        />
        Search
      </button>
    </form>
    <hr />
    <div
      className="row category-card"
    />
    <Pagination
      handlePageChange={[Function]}
      per_page={9}
      total=""
    />
  </div>
</div>
`;
